tests/integration/strategy_tests.cpp

```diff
@@ -470,15 +470,13 @@ TEST(ThreadSafeAllocator, HostStdThread)
   std::vector<void*> thread_allocs{N};
   std::vector<std::thread> threads;
 
-  for (std::size_t i = 0; i < N; ++i) {
-    threads.push_back(std::thread([=, &allocator, &thread_allocs] {
-      for (int j = 0; j < N; ++j) {
-        thread_allocs[i] = allocator.allocate(1024);
-        ASSERT_NE(thread_allocs[i], nullptr);
-        allocator.deallocate(thread_allocs[i]);
-        thread_allocs[i] = allocator.allocate(1024);
-        ASSERT_NE(thread_allocs[i], nullptr);
-      }
+  for (std::size_t i = 0; i < N; i++) {
+    threads.push_back(std::thread([=, &allocator, &thread_allocs] {  
```

Comment:
The intention of this test was to make certain that thread synchronization is working as expected for the allocator.  So the idea was to have each thread perform a large-ish (N) set of operations against the allocator with the hope that it would cause a lot of synchronization.  This change will cause the thread to only execute one sequence of allocate/deallocate.

Do we know why the original `for (int j = 0; j < N; ++j)) {` loop was causing a memory leak?

Reply:
I don't know for sure the exact cause, but I used ThreadSanitizer on this and found that this file had data races. I think that somewhere in those data races, a memory leak was occurring - and specifically on the second allocate call). Removing that loop solved the data races (except one minor one) and the leaks with it. This is documented in more detail on the JIRA ticket UM-678 (what the memory leak and data race errors were).

Reply:
Okay thanks @kab163, let's continue the discussion in UM-678.